docs/core/testing/unit-testing-mstest-writing-tests-testcontext.md

```diff
@@ -0,0 +1,183 @@
+---
+title: MSTest TestContext
+description: Learn about the TestContext class of MSTest.
+author: Evangelink
+ms.author: amauryleve
+ms.date: 11/12/2024
+---
+
+# The `TestContext` class
+
+The <xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestContext> class provides contextual information and support for test execution, making it easier to retrieve information about the test run and manipulate aspects of the environment. It's defined in the <xref:Microsoft.VisualStudio.TestTools.UnitTesting> namespace and is available when using the MSTest Framework.
+
+## Accessing the `TestContext` object
+
+The <xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestContext> object is available in the following contexts:
+
+- As a parameter to `[AssemblyInitialize]` and `[ClassInitialize]` methods. In this context, the properties related to the test run are not available.
+- As a property of a test class. In this context, the properties related to the test run are available.
+- As a constructor parameter of a test class (starting with v3.6) as an alternative to the property. This construct is recommended over the property because it gives access to the object in the constructor (while the property is only available after the constructor has run). It's also helping to ensure immutability of the object and finally it helps the compiler to enforce that the object is not null.
```

Comment:
```suggestion
- As a constructor parameter of a test class (starting with v3.6). This way is recommended over using the property, because it gives access to the object in the constructor. While the property is only available after the constructor has run. This way also helps to ensure immutability of the object and allows the compiler to enforce that the object is not null.
```